From 294123e632f3b935f4fad7e83ddcc74bc7e6eccf Mon Sep 17 00:00:00 2001
From: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
Date: Fri, 7 Dec 2018 07:46:39 +0800
Subject: [PATCH 397/706] switch: TSN: delete ssid interface from streamid set
 function

Using stream.handle ID as ssid and using stream.handle*2 as SFID
in stream identify function, so that only using one stream.handle
netlink interface to userspace.

Signed-off-by: Xiaoliang Yang <xiaoliang.yang_1@nxp.com>
(cherry picked from commit 8254176d4fbf4263de4581d83167e762d84aa654)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/tsn_switch.c | 11 ++++++-----
 include/uapi/linux/tsn.h               |  3 ---
 net/tsn/genl_tsn.c                     |  3 ---
 3 files changed, 6 insertions(+), 11 deletions(-)

diff --git a/drivers/net/ethernet/mscc/tsn_switch.c b/drivers/net/ethernet/mscc/tsn_switch.c
index 5a327a6b90bf..0ba6a4f5adc2 100644
--- a/drivers/net/ethernet/mscc/tsn_switch.c
+++ b/drivers/net/ethernet/mscc/tsn_switch.c
@@ -506,9 +506,9 @@ int switch_cb_streamid_set(struct net_device *ndev, u32 index, bool enable,
 
 	if (streamid->type == 1) {
 		if (enable == TRUE) {
-			netdev_dbg(ndev, "index=%d mac=0x%llx vid=0x%x ssid=%d sfid=%d dst=%d\n",
+			netdev_dbg(ndev, "index=%d mac=0x%llx vid=0x%x sfid=%d dst=%d\n",
 				   index, streamid->para.nid.dmac,
-				   streamid->para.nid.vid, streamid->ssid,
+				   streamid->para.nid.vid,
 				   streamid->handle,
 				   port->chip_port);
 
@@ -521,8 +521,8 @@ int switch_cb_streamid_set(struct net_device *ndev, u32 index, bool enable,
 					ANA_TABLES_MACHDATA_MACHDATA(mach),
 					ANA_TABLES_MACHDATA);
 
-			sfid = streamid->handle;
-			ssid = streamid->ssid;
+			sfid = streamid->handle * 2;
+			ssid = streamid->handle;
 			ocelot_write(ocelot,
 				     ((sfid >= 0) ? ANA_TABLES_STREAMDATA_SFID_VALID : 0) |
 				     ((sfid >= 0) ? ANA_TABLES_STREAMDATA_SFID(sfid) : 0) |
@@ -587,6 +587,7 @@ int switch_qci_sfi_set(struct net_device *ndev, u32 index, bool enable,
 	u16 sgid  = sfi->stream_gate_instance_id;
 	u16 pol_idx = sfi->stream_filter.flow_meter_instance_id;
 	u16 max_sdu_len = sfi->stream_filter.maximum_sdu_size;
+	int sfid = index * 2;
 
 	netdev_dbg(ndev, "sfid=%d prio=%d sgid=%d pol_idx=%d\n",
 		   index, igr_prio, sgid, pol_idx);
@@ -595,7 +596,7 @@ int switch_qci_sfi_set(struct net_device *ndev, u32 index, bool enable,
 		     ANA_TABLES_SFIDTIDX_SGID(sgid) |
 		     ANA_TABLES_SFIDTIDX_POL_ENA |
 		     ANA_TABLES_SFIDTIDX_POL_IDX(pol_idx) |
-		     ANA_TABLES_SFIDTIDX_SFID_INDEX(index),
+		     ANA_TABLES_SFIDTIDX_SFID_INDEX(sfid),
 		     ANA_TABLES_SFIDTIDX);
 
 	ocelot_write(ocelot,
diff --git a/include/uapi/linux/tsn.h b/include/uapi/linux/tsn.h
index 636e6f2d7ac9..d1cec731d49f 100644
--- a/include/uapi/linux/tsn.h
+++ b/include/uapi/linux/tsn.h
@@ -126,7 +126,6 @@ enum {
 	TSN_STREAMID_ATTR_ENABLE,
 	TSN_STREAMID_ATTR_DISABLE,
 	TSN_STREAMID_ATTR_STREAM_HANDLE,
-	TSN_STREAMID_ATTR_SSID,
 	TSN_STREAMID_ATTR_IFOP,
 	TSN_STREAMID_ATTR_OFOP,
 	TSN_STREAMID_ATTR_IFIP,
@@ -475,8 +474,6 @@ struct tsn_cb_streamid {
 	 */
 	int32_t handle;
 
-	int32_t ssid;
-
 	/* The list of ports on which an in-facing Stream identification function
 	 * in the output (towards the system forwarding function) direction
 	 * Only Active Destination MAC and VLAN Stream identification (or nothing)
diff --git a/net/tsn/genl_tsn.c b/net/tsn/genl_tsn.c
index fc079a0c36a4..ff6dff42a143 100644
--- a/net/tsn/genl_tsn.c
+++ b/net/tsn/genl_tsn.c
@@ -158,7 +158,6 @@ static const struct nla_policy cb_streamid_policy[TSN_STREAMID_ATTR_MAX + 1] = {
 	[TSN_STREAMID_ATTR_ENABLE] 	= { .type = NLA_FLAG},
 	[TSN_STREAMID_ATTR_DISABLE]	= { .type = NLA_FLAG},
 	[TSN_STREAMID_ATTR_STREAM_HANDLE]	= { .type = NLA_S32},
-	[TSN_STREAMID_ATTR_SSID]	= { .type = NLA_S32},
 	[TSN_STREAMID_ATTR_IFOP]	= { .type = NLA_U32},
 	[TSN_STREAMID_ATTR_OFOP]	= { .type = NLA_U32},
 	[TSN_STREAMID_ATTR_IFIP]	= { .type = NLA_U32},
@@ -530,8 +529,6 @@ static int cmd_cb_streamid_set(struct genl_info *info)
 	if (sid[TSN_STREAMID_ATTR_STREAM_HANDLE])
 		sidconf.handle = nla_get_s32(sid[TSN_STREAMID_ATTR_STREAM_HANDLE]);
 
-	if (sid[TSN_STREAMID_ATTR_SSID])
-		sidconf.ssid = nla_get_s32(sid[TSN_STREAMID_ATTR_SSID]);
 	if (sid[TSN_STREAMID_ATTR_IFOP])
 		sidconf.ifac_oport = nla_get_u32(sid[TSN_STREAMID_ATTR_IFOP]);
 	if (sid[TSN_STREAMID_ATTR_OFOP])
-- 
2.17.1

